src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1370,3 +1370,22 @@ def run_cmd(args, *, capture_output=False, timeout=None, check=False, encoding=N
     import subprocess
     return subprocess.run(args, capture_output=capture_output, timeout=timeout, check=check,
                           encoding=encoding, env=env)
+
+
+def run_az_cmd(args, out_file=None):
+    """
+    run_az_cmd would run az related cmds during command execution
+    :param args: cmd to be executed, array of string, like `["az", "version"]`, "az" is optional
+    :param out_file: The file to send output to. file-like object
+    :return: cmd execution result object, containing `result`, `error`, `exit_code`
+    """
+    from azure.cli.core.azclierror import ArgumentUsageError
+    if not isinstance(args, list):
+        raise ArgumentUsageError("Invalid args. run_az_cmd args must be a list")
+    if args[0] == "az":
+        args = args[1:]
+
+    from azure.cli.core import get_default_cli
+    cli = get_default_cli()
+    cli.invoke(args, out_file=out_file)
+    return cli.result
```

Comment:
If exception raised during cli.invoke, then no result can be fetched by caller but the exception itself.